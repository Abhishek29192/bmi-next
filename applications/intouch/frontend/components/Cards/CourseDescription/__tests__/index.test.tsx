import React from "react";
import { render } from "@testing-library/react";
import { CourseDescription } from "..";

describe("CourseDescription component", () => {
  it("renders correctly", () => {
    const { container } = render(
      <CourseDescription
        title="Basic competency programme (BCP)"
        type="Pitched roof"
        status="In progress"
        image="https://source.unsplash.com/MjLrM8rVMC0/799x500/?architecture"
      >
        <p>
          Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatem
          architecto placeat inventore nostrum voluptates deserunt possimus
          laborum autem eum? Cum temporibus debitis delectus veniam, deleniti
          sequi nobis ipsa explicabo voluptate!
        </p>
      </CourseDescription>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
