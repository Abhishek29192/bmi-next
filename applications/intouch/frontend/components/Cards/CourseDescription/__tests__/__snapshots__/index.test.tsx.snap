// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseDescription component renders correctly 1`] = `
<div
  class="main"
>
  <h4
    class="MuiTypography-root Typography Typography--underline MuiTypography-h4"
  >
    Basic competency programme (BCP)
  </h4>
  <div
    class="body"
  >
    <div
      class="header"
    >
      <div
        class="bannerImage"
      >
        <img
          alt="Basic competency programme (BCP)"
          src="https://source.unsplash.com/MjLrM8rVMC0/799x500/?architecture"
          style="width: 100%; display: block;"
        />
      </div>
      <div>
        <div
          class="metadata"
        >
          <div
            class="type"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
              />
            </svg>
            <h5
              class="MuiTypography-root Typography MuiTypography-h5"
            >
              Pitched roof
            </h5>
          </div>
          <div
            class="status"
          >
            In progress
          </div>
        </div>
        <div
          class="cta"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined Button MuiButton-outlinedPrimary"
            rel="noopener noreferrer"
            tabindex="0"
            target="_blank"
            type="button"
          >
            <span
              class="MuiButton-label"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 4l-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
                  />
                </svg>
              </span>
              View training
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
    <div
      class="body"
    >
      <p>
        Lorem ipsum dolor sit amet consectetur adipisicing elit. Voluptatem architecto placeat inventore nostrum voluptates deserunt possimus laborum autem eum? Cum temporibus debitis delectus veniam, deleniti sequi nobis ipsa explicabo voluptate!
      </p>
    </div>
  </div>
</div>
`;
